x-pack/plugins/security_solution/server/lib/detection_engine/signals/rule_status_saved_objects_client.ts

```diff
@@ -27,7 +28,7 @@ export interface RuleStatusSavedObjectsClient {
 }
 
 export const ruleStatusSavedObjectsClientFactory = (
-  savedObjectsClient: SavedObjectsClientContract
+  savedObjectsClient: SavedObjectsClientContract | SavedObjectsRepository
```

Comment:
I think you want `| ISavedObjectsRepository` instead and then you can remove the `as cast` as mentioned below?